### PR TITLE
Enhancements and improvements to DataTable feature: Delay Spinner, Simplified Sortable, Renamed DataTableButton, and More

### PR DESCRIPTION
**Pull request by patrickweh**
updated README.md

## Added
add ability to hide the table head
add table action buttons
add infinite scrolling [#5]
add interface to add methods to models
add ability to order by relation columns [#8]
add js function that tries to detect the formatter on a value

## CHANGED
spinner now has 500ms delay before showing.
use $id magic in case multiple tables on the same site exist
indexed arrays are no longer dotted.

**!!! BREAKING CHANGES !!!**
sortable is now just an array of keys.
renamed DataTableRowButton.php to DataTableButton.php

### CODE REVIEW

File: src/DataTable.php [hunk 1]
```diff
@@ -65,14 +74,46 @@ class DataTable extends Component
...
 
+    /**
+     * If set to false the table will have no head, so no captions for the cols.
+     *
+     * @locked
+     */
+    public bool $hasHead = true;
+
+    /**
+     * if set to true the table will show no pagination but
```
Comment (SirSplasch) [inline src/DataTable.php:106]:
```suggestion
     * If set to true the table will show no pagination but
```

File: src/DataTable.php [hunk 2]
```diff
@@ -444,22 +506,36 @@ public function getReturnKeys(): array
...
+        foreach ($returnKeys as $key) {
+            $value = $itemArray[$key] ?? false;
+            if ($value && is_array($value) && ! Arr::isAssoc($value)) {
+                $preserved[$key] = Arr::pull($itemArray, $key);
+            }
+        }
+
+        $itemArray = Arr::only(array_merge(Arr::dot($itemArray), $preserved), $returnKeys);
+        $itemArray['href'] = in_array(InteractsWithDataTables::class, class_implements($this->model))
+        && ! $this->hasNoRedirect
```
Comment (SirSplasch) [inline src/DataTable.php:526]: sauber formatieren

File: src/DataTable.php [hunk 3]
```diff
@@ -659,10 +735,42 @@ public function buildSearch(): Builder
...
+                }
+
+                $table = $relationInstance->getRelated()->getTable();
+                $first = $relationInstance instanceof BelongsTo ? $relationInstance->getQualifiedOwnerKeyName() : $relationInstance->getQualifiedParentKeyName();
+                $second = $relationInstance->getQualifiedForeignKeyName();
+
+                $query->join($table, $first, '=', $second)->where($second, '!=', null);
+
+                $localModel = $relationInstance->getRelated();
+            }
```
Comment (SirSplasch) [inline src/DataTable.php:766]: Leerzeile zwischenrein

File: src/DataTable.php [hunk 4]
```diff
@@ -659,10 +735,42 @@ public function buildSearch(): Builder
...
-            $query->orderBy($this->orderBy, $this->orderAsc ? 'asc' : 'desc');
+        if (Str::contains($this->orderBy, '.')) {
+            $relationPath = explode('.', $this->orderBy);
+            $orderByColumn = array_pop($relationPath);
+            $localModel = new $model;
+            $query->addSelect($localModel->getTable() . '.*');
+
+            foreach ($relationPath as $key => $relation) {
+                $class = new ReflectionClass($localModel);
+                /** @var HasOne|BelongsTo $relationInstance */
```
Comment (SirSplasch) [inline src/DataTable.php:746]: returns relation instance

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/DataTable.php (hunk 3 above)
Comment (patrickweh) [reply src/DataTable.php:768]: √

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**